### PR TITLE
added trid objective function

### PR DESCRIPTION
**Pull request by hstojic**
objective function from Hebbal et al 2019 (https://arxiv.org/abs/1905.03350), vanilla GPs seem to perform poorly, while DGPs well
might also be good testing ground for data transformation/normalisation (if and when we come to that)

### CODE REVIEW

File: docs/refs.bib [hunk 1]
```diff
@@ -215,3 +215,9 @@ @article{hebbal2019bayesian
   journal={arXiv preprint arXiv:1905.03350},
   year={2019}
 }
+
+@misc{ssurjano2021, 
+ author = {Surjanovic, S. and Bingham, D.}, 
+ title = {Virtual Library of Simulation Experiments: Test Functions and Datasets}, 
+ howpublished = {Retrieved October 15, 2021, from \url{http://www.sfu.ca/~ssurjano}} 
+}
```
Comment (apaleyes) [inline docs/refs.bib]: Other citations have two spaces indent

File: trieste/objectives/single_objectives.py [hunk 2]
```diff
@@ -511,3 +511,50 @@ def michalewicz_10(x: TensorType) -> TensorType:
 """
 The search space for the 10-dimensional :func:`michalewicz` function.
 """
+
+
+def trid(x: TensorType, d: int = 10) -> TensorType:
+    """
+    The Trid function over :math:`[-d^2, d^2]` for all i=1,...,d. Dimensionality is determined
+    by the parameter ``d`` and it has a global minimum. This functions has large variation in
```
Comment (apaleyes) [inline trieste/objectives/single_objectives.py]:
```suggestion
    by the parameter ``d`` and it has a global minimum. This function has large variation in
```

File: trieste/objectives/single_objectives.py [hunk 3]
```diff
@@ -511,3 +511,50 @@ def michalewicz_10(x: TensorType) -> TensorType:
...
+    output which makes it challenging for Bayesian optimisation with vanilla Gaussian processes
+    with non-stationary kernels. Models that can deal with non-stationarities, such as deep
+    Gaussian processes, can be useful for modelling these functions. See :cite:`hebbal2019bayesian`
+    and https://www.sfu.ca/~ssurjano/trid.html for details.
```
Comment (apaleyes) [inline trieste/objectives/single_objectives.py:523]: Maybe insert citation instead of an url here?

**Review by apaleyes — APPROVED** [review]
Just minor comments

File: trieste/objectives/single_objectives.py (hunk 3 above)
Comment (hstojic) [reply trieste/objectives/single_objectives.py:523]: url specifically points to trid part of the website

**Pull request merged**